lightning/src/ln/channel.rs

```diff
@@ -2689,12 +2731,10 @@ impl<Signer: Sign> Channel<Signer> {
 			}
 		}
 		for htlc in self.pending_outbound_htlcs.iter_mut() {
-			if let Some(fail_reason) = if let &mut OutboundHTLCState::RemoteRemoved(ref mut fail_reason) = &mut htlc.state {
-				Some(fail_reason.take())
-			} else { None } {
+			if let &OutboundHTLCState::RemoteRemoved(ref fail_reason) = &htlc.state {
 				log_trace!(logger, "Updating HTLC {} to AwaitingRemoteRevokeToRemove due to commitment_signed in channel {}.",
 					log_bytes!(htlc.payment_hash.0), log_bytes!(self.channel_id));
-				htlc.state = OutboundHTLCState::AwaitingRemoteRevokeToRemove(fail_reason);
+				htlc.state = OutboundHTLCState::AwaitingRemoteRevokeToRemove(fail_reason.clone().into());
```

Comment:
Yuck. Can we `mem::swap` with a dummy value to avoid cloning Vecs just to drop? Same below.

Reply:
fixed